code/_onclick/hud/radial.dm

```diff
@@ -273,6 +270,12 @@ GLOBAL_LIST_EMPTY(radial_menus)
 /datum/radial_menu/Destroy()
 	Reset()
 	hide()
+	for(var/obj/screen/radial/slice/slice_to_remove in elements)
+		slice_to_remove.parent = null
```

Comment:
This should be done in the `Destroy` of `/obj/screen/radial`. This also allows you to use `QDEL_LIST_CONTENTS`

Reply:
Good idea, I'll implement it when I get onto my PC (eta 2 hours)

Reply:
Done